spec/abstractops.html

```diff
@@ -196,51 +196,54 @@ <h1>ToShowOffsetOption ( _normalizedOptions_ )</h1>
         _normalizedOptions_: an Object,
-        _dividend_: an integer or *undefined*,
+      ): either a normal completion containing a Number, or an abrupt completion
+    </h1>
+    <dl class="header">
+      <dt>description</dt>
+      <dd>
+        It extracts the value of the property named *"roundingIncrement"* from _normalizedOptions_, makes sure it is a finite Number greater than or equal to 1, and returns that value.
+        It performs no further validation.
+      </dd>
+    </dl>
+    <emu-alg>
+      1. Let _increment_ be ? GetOption(_normalizedOptions_, *"roundingIncrement"*, *"number"*, *undefined*, *1*<sub>𝔽</sub>).
+      1. If _increment_ is not finite, throw a *RangeError* exception.
+      1. If _increment_ &lt; *1*<sub>𝔽</sub>, throw a *RangeError* exception.
+      1. Return _increment_.
+    </emu-alg>
+  </emu-clause>
+
+  <emu-clause id="sec-validatetemporalroundingincrement" type="abstract operation">
+    <h1>
+      ValidateTemporalRoundingIncrement (
+        _increment_: a Number,
+        _dividend_: a positive integer,
         _inclusive_: a Boolean,
-      )
+      ): either a normal completion containing an integer, or an abrupt completion
     </h1>
     <dl class="header">
       <dt>description</dt>
       <dd>
-        It extracts the value of the property named *"roundingIncrement"* from _normalizedOptions_, makes sure it is a valid value for the option, and returns that value rounded to an integer.
-        If _dividend_ is *undefined*, any non-zero positive value is valid.
-        Otherwise, the value must not be more than _dividend_ (or must be less than _dividend_, if _inclusive_ is *false*) and must divide evenly into _dividend_.
+        It performs further validation of a value _increment_ returned from ToTemporalRoundingIncrement.
+        It checks whether _increment_ divides evenly into _dividend_, and throws an exception if not.
+        If _inclusive_ is *true*, _increment_ is allowed to be equal to _dividend_, otherwise it must be less.
       </dd>
     </dl>
     <emu-alg>
-      1. If _dividend_ is *undefined*, then
-        1. Let _maximum_ be *+&infin;*<sub>𝔽</sub>.
-      1. Else if _inclusive_ is *true*, then
+      1. If _inclusive_ is *true*, then
         1. Let _maximum_ be 𝔽(_dividend_).
       1. Else if _dividend_ is more than 1, then
         1. Let _maximum_ be 𝔽(_dividend_ - 1).
       1. Else,
         1. Let _maximum_ be *1*<sub>𝔽</sub>.
-      1. Let _increment_ be ? GetOption(_normalizedOptions_, *"roundingIncrement"*, *"number"*, *undefined*, *1*<sub>𝔽</sub>).
-      1. If _increment_ &lt; *1*<sub>𝔽</sub> or _increment_ &gt; _maximum_, throw a *RangeError* exception.
-      1. Set _increment_ to floor(ℝ(_increment_)).
-      1. If _dividend_ is not *undefined* and _dividend_ modulo _increment_ is not zero, then
+      1. If _increment_ &gt; _maximum_, throw a *RangeError* exception.
+      1. Set _increment_ to truncate(ℝ(_increment_)).
```

Comment:
What motivates changing `floor(ℝ(_increment_))` to `truncate(ℝ(_increment_))`, given that any value returned from ToTemporalRoundingIncrement is positive?

Also, is it relevant that _increment_ is compared to _maximum_ **before** rounding? I'm concerned about propagating the problems highlighted by https://github.com/tc39/ecma402/issues/691 .

Reply:
I just generally reach for truncate over floor out of habit, but you're right, there's no reason to change the existing use of floor here. I'll change it back.

I had to keep the order of comparison vs rounding unchanged here because it would otherwise be a normative change, but I made the normative change in #2437 which achieved consensus today.